### PR TITLE
Integrate firebase into CI for examples

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes
This PR introduces Fastlane to sign app and run the example test on firebase instead of aws.

### CODE REVIEW

File: .circleci/config.yml
Comment (Chaoba) [inline .circleci/config.yml:110]: This need removed after review

File: fastlane/Matchfile
Comment (OdNairy) [inline fastlane/Matchfile:1]: This is mostly a duplicate of `/Apps/.fastlane/Matchfile`

File: Apps/.fastlane/Fastfile
Comment (OdNairy) [inline Apps/.fastlane/Fastfile:39]: It looks like we do not gather `xcresult` in this action. Documentation says we have to use `result_bundle: true`  to generate one in the output folder.

File: .circleci/config.yml
Comment (OdNairy) [inline .circleci/config.yml:467]: Did you prune this action because of missing `xcresult`? If so, we should recover it as soon as we add `result_bundle` to the `scan` action.

**Review by OdNairy — CHANGES_REQUESTED** [review]
Great job, Kevin! Glad to see Firebase integration working.
We still need some fixes and cleanup. I would be happy to re-review after new commits would be landed.

File: .circleci/config.yml
Comment (Chaoba) [reply .circleci/config.yml:467]: The CI will give the output uri of firebase, we can check the failed result with this uri directly. This is the way we are using for Android too.

File: fastlane/Matchfile
Comment (Chaoba) [reply fastlane/Matchfile]: Yeah, deleted this folder

File: Apps/.fastlane/Matchfile
Comment (Chaoba) [reply Apps/.fastlane/Matchfile]: Gl-native choose to set it in env, let's algin with them and it's safer.

File: Apps/.fastlane/Appfile
Comment (Chaoba) [reply Apps/.fastlane/Appfile:2]: These are commented out, guess we don't need them.

File: .circleci/config.yml
Comment (Chaoba) [reply .circleci/config.yml:804]: Yes, deleted

File: .circleci/config.yml
Comment (Chaoba) [reply .circleci/config.yml:641]: Yes,  that's the project in firebase.  Do you think it's unsafe if we use it directly?

File: Apps/Examples/Examples/All Examples/CameraAnimationExample.swift
Comment (OdNairy) [inline Apps/Examples/Examples/All Examples/CameraAnimationExample.swift:22]: Was that change intentional?

File: .circleci/config.yml
Comment (OdNairy) [reply .circleci/config.yml:641]: I think it is just more configurable. We would be able to change the firebase project without any changes in the source code. 
My general motivation is that it is okay to hardcode Apple related information (including team-wide like applemachine account name or team identifier) but I would like to avoid any unusual constants like project identifier.

File: Apps/.fastlane/Matchfile
Comment (OdNairy) [reply Apps/.fastlane/Matchfile]: Related: https://github.com/mapbox/mapbox-maps-ios/pull/914#discussion_r774448866

File: .circleci/config.yml
Comment (OdNairy) [reply .circleci/config.yml:467]: @macdrevx Do we still need `xcparty` tool when working with Firebase?

File: Apps/Examples/Examples/All Examples/CameraAnimationExample.swift
Comment (Chaoba) [reply Apps/Examples/Examples/All Examples/CameraAnimationExample.swift:22]: Yes, to trigger a crash, will revert it when everything goes well

File: .circleci/config.yml
Comment (Chaoba) [reply .circleci/config.yml:641]: Will extract it to environment variable.

File: .circleci/config.yml
Comment (Chaoba) [reply .circleci/config.yml]: Build with release will have error: `'OfflineRegionManagerExample' is deprecated`, so will keep use Debug for testing

File: Makefile
Comment (OdNairy) [inline Makefile:267]: It looks like we have double build-for-testing logic – first here and the second in Fastfile. Do we need them both?

File: .circleci/config.yml
Comment (OdNairy) [inline .circleci/config.yml:771]:
```suggestion
            zip -r "$ZIP_FILE" test_result_0.xml || echo "Skipping test_result_0.xml"
```

File: .circleci/config.yml
Comment (OdNairy) [inline .circleci/config.yml:777]:
```suggestion
          when: always
```

File: .circleci/config.yml
Comment (OdNairy) [inline .circleci/config.yml:790]: Can we extract google-cloud-sdk version and checksum as command parameters with default values? It would be easier to update them later.

Comment (OdNairy) [reply .circleci/config.yml:790]: Can we also leave a link to the documentation where we can find hashsum for a newer version?

File: Makefile
Comment (Chaoba) [reply Makefile]: Removed this part.

File: Makefile
Comment (OdNairy) [inline Makefile:271]: We can replace this two commands with additional argument to the `scan` fastlane action – `should_zip_build_products`

File: Makefile
Comment (OdNairy) [inline Makefile:275]: Can we consider the usage of the Fastlane plugin – https://github.com/fastlane/fastlane-plugin-firebase_test_lab ? 
It would simplify firebase testing to a single fastlane call. Also, the full configuration would live in the single Fastfile.

File: .circleci/config.yml
Comment (OdNairy) [inline .circleci/config.yml:431]:
```suggestion
      - run:
          name: Start Firebase lane
          working_directory: Apps
          command: bundle install && bundle exec fastlane firebase
```

File: .circleci/config.yml
Comment (macdrevx) [reply .circleci/config.yml:467]: I think that tool was used to parse the results from device farm and fail the Circle CI job if it found test failures. I assume we'd want something similar with firebase.

File: .circleci/config.yml
Comment (macdrevx) [inline .circleci/config.yml:814]: Seems like this is unused.

File: Apps/.fastlane/Fastfile
Comment (macdrevx) [inline Apps/.fastlane/Fastfile:14]:
```suggestion
```

**Review by macdrevx — COMMENTED** [review]
Looking good. I defer to @OdNairy for final approval. Excited to see it in action.

File: Makefile
Comment (Chaoba) [reply Makefile:275]: As discussed, will iterate this in the future.

**Review by OdNairy — APPROVED** [review]
I think we can approve it.
However, we still missing two important features:
 - [ ] Pass device matrixes as we had in Device Farm
 - [ ] Retrieve and parse XCResult bundle

**Pull request merged**
